tools/accgen/accgen.sh

```diff
@@ -598,6 +697,34 @@ if [ "$FLOW" == "stratus_hls" ]; then
     fi
 fi
 
+### Generate skeleton for unit testbench
+if [ "$FLOW" == "catapult_hls" ]; then
+    cd $ACC_DIR/hw/inc
+    # accelerator_specs.hpp
+    sed -i "s/\/\* <<--mem-footprint-->> \*\//${memory_footprint}/g" ${LOWER}_specs.hpp
+    sed -i "/\/\* <<--defines-->> \*\//a #define DATA_WIDTH $data_width" ${LOWER}_specs.hpp
+    sed -i "/\/\* <<--defines-->> \*\//a #define PLM_IN_WORD $in_word" ${LOWER}_specs.hpp
+    sed -i "/\/\* <<--defines-->> \*\//a #define PLM_OUT_WORD $out_word" ${LOWER}_specs.hpp
+
+    cd $ACC_DIR/hw/tb
+    # testbench.hpp
+    sed -i "s/\/\* <<--data_out_size-->> \*\//${data_out_size_expr}/g" testbench.hpp
+    for key in ${!values[@]}; do
+	indent="\ \ \ \ "
+	sed -i "/\/\* <<--params-->> \*\//a ${indent}uint32_t ${key};" testbench.hpp
+	indent="\ \ \ \ \ \ \ \ "
+	sed -i "/\/\* <<--params-default-->> \*\//a ${indent}${key} = ${values[$key]};" testbench.hpp
+    done
+    # testbench.cpp
+    sed -i "s/\/\* <<--number of transfers-->> \*\//${batching_factor_expr}/g" testbench.cpp
+    sed -i "s/\/\* <<--data_in_size-->> \*\//${data_in_size_expr}/g" testbench.cpp
+    sed -i "s/\/\* <<--data_out_size-->> \*\//${data_out_size_expr}/g" testbench.cpp
+    for key in ${!values[@]}; do
+	indent="\ \ \ \ \ \ \ \ "
+	sed -i "/\/\* <<--params-->> \*\//a ${indent}config.${key} = ${key};" testbench.cpp
+    done
+fi
+
 
```

Comment:
have you tested accgen with the other flow since your modifications? please check stratus at least